lib/roo/excelx.rb

```diff
@@ -1,667 +1,491 @@
-require 'date'
 require 'nokogiri'
 require 'roo/link'
 require 'roo/utils'
 require 'zip/filesystem'
 
-class Roo::Excelx < Roo::Base
-  autoload :Workbook, 'roo/excelx/workbook'
-  autoload :SharedStrings, 'roo/excelx/shared_strings'
-  autoload :Styles, 'roo/excelx/styles'
-
-  autoload :Relationships, 'roo/excelx/relationships'
-  autoload :Comments, 'roo/excelx/comments'
-  autoload :SheetDoc, 'roo/excelx/sheet_doc'
-
-  module Format
-    EXCEPTIONAL_FORMATS = {
-      'h:mm am/pm' => :date,
-      'h:mm:ss am/pm' => :date,
-    }
-
-    STANDARD_FORMATS = {
-      0 => 'General',
-      1 => '0',
-      2 => '0.00',
-      3 => '#,##0',
-      4 => '#,##0.00',
-      9 => '0%',
-      10 => '0.00%',
-      11 => '0.00E+00',
-      12 => '# ?/?',
-      13 => '# ??/??',
-      14 => 'mm-dd-yy',
-      15 => 'd-mmm-yy',
-      16 => 'd-mmm',
-      17 => 'mmm-yy',
-      18 => 'h:mm AM/PM',
-      19 => 'h:mm:ss AM/PM',
-      20 => 'h:mm',
-      21 => 'h:mm:ss',
-      22 => 'm/d/yy h:mm',
-      37 => '#,##0 ;(#,##0)',
-      38 => '#,##0 ;[Red](#,##0)',
-      39 => '#,##0.00;(#,##0.00)',
-      40 => '#,##0.00;[Red](#,##0.00)',
-      45 => 'mm:ss',
-      46 => '[h]:mm:ss',
-      47 => 'mmss.0',
-      48 => '##0.0E+0',
-      49 => '@',
-    }
-
-    def to_type(format)
-      format = format.to_s.downcase
-      if type = EXCEPTIONAL_FORMATS[format]
-        type
-      elsif format.include?('#')
-        :float
-      elsif !format.match(/d+(?![\]])/).nil? || format.include?('y')
-        if format.include?('h') || format.include?('s')
-          :datetime
+module Roo
+  class Excelx < Roo::Base
+    autoload :Workbook, 'roo/excelx/workbook'
```

Comment:
I think we can require this modules except of autoload. No need to make autoload there.